zenith_utils/src/zid.rs

```diff
@@ -167,34 +215,6 @@ pub struct ZTenantId(ZId);
 
 zid_newtype!(ZTenantId);
 
-/// Serde routines for Option<T> (de)serialization, using `T:Display` representations for inner values.
-/// Useful for Option<ZTenantId> and Option<ZTimelineId> to get their hex representations into serialized string and deserialize them back.
-pub mod opt_display_serde {
```

Comment:
Could you please elaborate, how manual serialization of zid helps to avoid problem with option?

Reply:
It's all possible due to `Visitor` interface.

Before that, various `#[serde(with = "hex")]` and the custom one I've removed were needed, separate for every type: if you run the new test on main, you would see that all ZIds were serialized as arrays instead of readable hex strings.

Instead of adding yet another option-like stub deserializer for a particular type, I've decided to implement a proper deserializer for ZId itself instead, so no "hex" or anything is needed.

`Visitor` is called recursively on types (via serde), allowing its serializers (json, in our case), to apply logic on top.
Say, this visitor in case of some hex string value for `Option<ZId>` will first attempt to call visitor's method for option, will get nothing from the default impl and then (knowing JS semantics), will attempt to parse the str/bytes as the ZId itself, which will work.

More here: https://serde.rs/impl-deserializer.html

-----
This approach will, in theory, enable proper ZId deserialization for every type and every serde-based impl, yet there are 3 tests that get stuck infinitely somewhere around `callmemaybe` and I'm not sure what happens.

Most probably, some custom (byte based?) deserialization fails/gets stuck due to ZId's Visitor not implementing something.

Reply:
Great! Thanks for the explanation :)

> and I'm not sure what happens.

Ugh, thats weird. Which particular tests get stuck?

Reply:
[test_next_xid](https://app.circleci.com/pipelines/github/zenithdb/zenith/4702/workflows/36a8f206-6580-4a1e-8bfc-b52a05824723/jobs/44475/parallel-runs/0/steps/0-106)